roles/register_hypervisor/tasks/main.yml

```diff
@@ -12,21 +12,31 @@
 
 - name: copy hypervisor registration script
   copy:
-    src: files/add_host
+    src: 'files/add_host{{ (compatibility_version < 4.1) | ternary("","_sdk4") }}'
     dest: '{{ config_dir }}/scripts/add_host'
     mode: 0700
   delegate_to: '{{ script_runner }}'
   tags:
     - config
 
+- name: Install python-ovirt-engine-sdk4
+  yum:
+    name: python-ovirt-engine-sdk4
+    state: present
+  when: compatibility_version >= 4.1
```

Comment:
You're going to want to delegate this task to the script runner, otherwise it might not be installed on the host that actually runs it.